plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -144,6 +169,21 @@ public S3FileSystemExchangeStorage(S3FileSystemExchangeStorageStats stats, Excha
                 config.getAsyncClientConcurrency(),
                 config.getAsyncClientMaxPendingConnectionAcquires(),
                 config.getConnectionAcquisitionTimeout());
+
+        if (compatibilityMode.equals(GCP)) {
+            if (config.getGcsKeyPath().isPresent()) {
+                Credentials credentials = GoogleCredentials.fromStream(new FileInputStream(config.getGcsKeyPath().get()));
+                this.gcsClient = Optional.of(StorageOptions.newBuilder().setCredentials(credentials).build().getService());
+            }
+            else {
+                this.gcsClient = Optional.of(StorageOptions.getDefaultInstance().getService());
+            }
+            this.gcsDeleteExecutor = MoreExecutors.listeningDecorator(newCachedThreadPool(threadsNamed("gcs-delete-%s")));
```

Comment:
This can in theory grow indefinitely. In `FileSystemExchange#close` we call `deleteRecursively` in a loop for each task. This may result in huge spikes in number of threads (hundreds or even thousands). I would recommend going with a bounded executor with the number of threads set to desired concurrency, e.g.:

```
        ThreadPoolExecutor executor = new ThreadPoolExecutor(
                maximumConcurrency,
                maximumConcurrency,
                10, SECONDS,
                new LinkedBlockingQueue<>(),
                threadsNamed("gcs-delete-%s"));
```

I don't know what value do we want to pick for max concurrency though, maybe 50? 100?

Also it is a generally good idea to set `executor.allowCoreThreadTimeOut(true);` to let the inactive threads be reclaimed after a spike. 

It might also be reasonable to improve `FileSystemExchange#close` to batch delete requests across multiple partitions.